Remove unused api.js

millerf opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/pull/32095#issuecomment-1663901576

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Use store to retrieve the subscribers instead of a API call.
* Clean up some files and reuse some logic 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to your favorite testing site 
* Add a subscribe button to your post
* See this in the panel. 🚀
<img width="300" alt="Screenshot 2023-08-04 at 13 53 15" src="https://github.com/Automattic/jetpack/assets/790558/9b97cc82-80fd-4cdd-9853-b68cfdceb7a3">

millerf committed:
[not verified] Remove api.js

millerf committed:
[not verified] Create remove-unused-api-js

millerf committed:
[not verified] Remove jest mock

millerf committed:
[not verified] Move useSelect before any other logic

millerf committed:
[not verified] Fix some logic

millerf committed:
[not verified] re-trigger tests

millerf committed:
[not verified] Try to fix tests

chrismccluskey commented on projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js:
```diff
@@ -72,6 +72,31 @@ export function SubscriptionEdit( props ) {
 	const { isLoadingModules, isChangingStatus, isModuleActive, changeStatus } =
 		useModuleStatus( name );
+
+	const { subscriberCount, subscriberCountString } = useSelect( select => {
```
I'm not sure why, but my local build of this branch as well as a build on JN (and using this branch directly with Jetpack Beta) both seem to think that `useSelect` is not a function, preventing this from working. Specifically, this happens when adding a subscribe button or viewing a page with a subscribe button. The result (along with the console log) is that the block isn't rendered. 

I see that we're importing it above, but it doesn't seem to be working. Do we want to use `withSelect` here? I see that it's already imported although I'm not sure of the difference between the two currently. I tried `withSelect` locally and it seemed to work as I would expect `useSelect` to work here.

millerf replied:
> I see that we're importing it above, but it doesn't seem to be working. Do we want to use withSelect here? I see that it's already imported although I'm not sure of the difference between the two currently. I tried withSelect locally and it seemed to work as I would expect useSelect to work here.

There might be a conflict using both approaches at the same time... But [it is not in the doc](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useregistry)...

millerf replied:
@chrismccluskey The issue ended up being `useSelect` was imported from the wrong package (`WP/elements` instead of `WP/data`).

But in the meantime I tried the `withSelect` approach that could not be used because of `isModuleActive` being retrieved from `useModuleStatus`. But you can have a look at the implementation on d0dd537da4

chrismccluskey reviewed (changes requested):
I left a comment on one small change. I'm also seeing that some check annotations in unchanged files seem to be failing (see "Unchanged files with check annotations"). I'm not entirely sure what I'm looking at here but thought I'd point that out.

millerf committed:
move to withSelect paradigm

millerf committed:
[not verified] Revert "move to withSelect paradigm"

This reverts commit d0dd537da4df62f3dda678712dd68d2d6bfbd925.

millerf committed:
Fix useSelect

millerf committed:
Fix issue

- useSelect was imported from the wrong package!!!
- mistake in spelling `subscriberCounts`
- `includeSocialFollowers` was defined after usage

chrismccluskey reviewed (approved):
This works now, LGTM!

jeherve reviewed (approved):
LGTM! 🚢
